src/Paramore.Brighter.MessagingGateway.AWSSQS/SnsMessageProducer.cs

```diff
@@ -0,0 +1,177 @@
+        if (!string.IsNullOrEmpty(ChannelTopicArn)) return !string.IsNullOrEmpty(ChannelTopicArn);
+
+        RoutingKey? routingKey = null;
+        if (topic is null && _publication.Topic is not null)
+            routingKey = _publication.Topic;
+        else if (topic is not null)
+            routingKey = new RoutingKey(topic);
+
+        if (routingKey is null)
+            throw new ConfigurationException("No topic specified for producer");
+
+        await EnsureTopicAsync(
+            routingKey,
+            _publication.FindTopicBy,
+            _publication.SnsAttributes,
+            _publication.MakeChannels,
+            _publication.SnsType,
+            _publication.Deduplication,
+            cancellationToken);
+
+        return !string.IsNullOrEmpty(ChannelTopicArn);
+    }
+
+    /// <summary>
+    /// Sends the specified message.
+    /// </summary>
+    /// <param name="message">The message.</param>
+    /// <param name="cancellationToken">Allows cancellation of the Send operation</param>
+    public async Task SendAsync(Message message, CancellationToken cancellationToken = default)
+    {
+        s_logger.LogDebug(
+            "SQSMessageProducer: Publishing message with topic {Topic} and id {Id} and message: {Request}",
+            message.Header.Topic, message.Id, message.Body);
+
+        await ConfirmTopicExistsAsync(message.Header.Topic, cancellationToken);
+
+        if (string.IsNullOrEmpty(ChannelTopicArn))
+            throw new InvalidOperationException(
+                $"Failed to publish message with topic {message.Header.Topic} and id {message.Id} and message: {message.Body} as the topic does not exist");
+
+        using var client = _clientFactory.CreateSnsClient();
+        var publisher = new SqsMessagePublisher(ChannelTopicArn!, client, _publication.SnsType, _publication.Deduplication);
+        var messageId = await publisher.PublishAsync(message);
+
+        if (messageId == null)
+            throw new InvalidOperationException(
+                $"Failed to publish message with topic {message.Header.Topic} and id {message.Id} and message: {message.Body}");
+
+        s_logger.LogDebug(
+            "SQSMessageProducer: Published message with topic {Topic}, Brighter messageId {MessageId} and SNS messageId {SNSMessageId}",
+            message.Header.Topic, message.Id, messageId);
+    }
+
+    /// <summary>
+    /// Sends the specified message.
+    /// Sync over Async
+    /// </summary>
+    /// <param name="message">The message.</param>
+    public void Send(Message message)
+    {
+        // TODO: Uncomment when the BrighterSynchronizationHelper is fixed, today the code isn't working because it's stuck is wired infinity loop
+        // BrighterSynchronizationHelper.Run(async () => await SendWithDelayAsync(message, delay));
+        SendAsync(message).GetAwaiter().GetResult();
```

Comment:
it is possible we could flip these back to the BrighterAsyncContext.Run, but perhaps another PR :-D